pytorch_lightning/profiler/base.py

```diff
@@ -113,14 +113,20 @@ def _rank_zero_info(self, *args, **kwargs) -> None:
         if self._local_rank in (None, 0):
             log.info(*args, **kwargs)
 
+    def _add_suffix(self, filename: str, suffix: Optional[str], token: str = '-') -> str:
+        if suffix is None:
+            return filename
+
+        if len(filename) >= 1 and filename[-1] != token:
+            filename += token
+        return filename + suffix
+
     def _prepare_filename(self, extension: str = ".txt") -> str:
         filename = ""
-        if self._stage is not None:
-            filename += f"{self._stage}-"
-        filename += str(self.filename)
-        if self._local_rank is not None:
-            filename += f"-{self._local_rank}"
-        filename += extension
+        filename = self._add_suffix(filename, self._stage.value if self._stage else None)
+        filename = self._add_suffix(filename, self.filename if self.filename else None)
+        filename = self._add_suffix(filename, str(self._local_rank) if self._local_rank is not None else None)
+        filename += extension if extension else ''
```

Comment:
why the if here? It should be always a str

```suggestion
        filename += extension
```

Reply:
Done !